Homework4: Maksim Kvach

nkvamaks opened the pull request:
Seven tasks were created alongside with two text files:
test_cities.txt and test_languages.txt that contain input
data for corresponding tasks.

### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle)
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

nkvamaks committed:
Initial commit

Seven tasks were created alongside with two text files:
test_cities.txt and test_languages.txt that contain input
data for corresponding tasks.

nkvamaks committed:
Fix output in 2_cities.py

belovmd commented on src/homework4/2_cities.py:
```diff
@@ -0,0 +1,28 @@
+    for _N in range(int(f.readline())):
+        country, *cities = f.readline().split()
+        for city in cities:
+            relation[city] = relation.get(city, []) + [country]
```
append?

nkvamaks replied:
Using 'append' was my first idea but it doesn't work in this situation.

belovmd replied:
it should. this code looks weird

belovmd commented on src/homework4/7_euclid_gcd.py:
```diff
@@ -0,0 +1,21 @@
+second_int = int(input("Input 2nd natural number: "))
+first_saved, second_saved = first_int, second_int
+
+if first_int < second_int:
```
unclear naming

nkvamaks replied:
Introduced clearer naming.

belovmd reviewed (commented):
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework4.md):

- redundant comments in code
- 21

Overall it's the best work I've seen from your group.

nkvamaks commented:
Thank you for your review.
I've addressed your comments as much as I could: really redundant comments were deleted, but I've left some comments alongside with docstrings to clarify blocks of code. For you as a professional programmer these comments may still seem like redundant but for me as a beginner they remind what idea is inside the block of code, so please be gentle.
Task 4 was fixed.

nkvamaks committed:
Update all tasks

nkvamaks committed:
Get rid of ALL 'redundant comments'

nkvamaks commented:
Just in case all 'redundant comments' were removed from code... Have no idea why you hate them so much...
Ngā mihi.

belovmd commented:
> Just in case all 'redundant comments' were removed from code... Have no idea why you hate them so much...
> Ngā mihi.

If there are redundant comments in code - it's bad practice
If they are your temporary comments for yourself - it's just inaccurate

belovmd commented on src/homework4/4_num_lists.py:
```diff
@@ -0,0 +1,13 @@
+
+set1 = set(list1)
+set2 = set(list2)
+print(len(set1.symmetric_difference(set2)))
```
also ^

belovmd reviewed (commented):
9,5
